### DISCUSSION TITLE
Performing the partial derivative of a MX function with respect to another one

### DISCUSSION THREAD

**Discussion by gavilomon** [casadi/casadi#3545] [Q&A]
Hello!

I am trying to perform the partial derivative of a function with respect to another function I evaluate within the discretisation loop to formulate the collocation constraints. 

So basically I create the functions before the discretisation loop with the Function command and then evaluate them within the loop with the Xk and Uk collocation variables in order to get the value of the functions at the collocation points. These yield two MX objects and I was wondering if there was a way to perform the partial derivative of one function with respect to the other (in order to subject it to a constraint afterwards). I have tried using the command jacobian() but doesn't seem to work here.

Many thanks!

**Comment by jgillis** [comment]
Can you provide a simple example along with expected numerical or symbolix output?
